      <!DOCTYPE html>
      <html>
        <head>
            <meta charset="utf-8" />
            <title>README</title>
            <style>.markdown-preview:not([data-use-github-style]) { padding: 2em; font-size: 1.2em; color: rgb(85, 85, 85); background-color: white; overflow: auto; }
.markdown-preview:not([data-use-github-style]) > :first-child { margin-top: 0px; }
.markdown-preview:not([data-use-github-style]) h1, .markdown-preview:not([data-use-github-style]) h2, .markdown-preview:not([data-use-github-style]) h3, .markdown-preview:not([data-use-github-style]) h4, .markdown-preview:not([data-use-github-style]) h5, .markdown-preview:not([data-use-github-style]) h6 { line-height: 1.2; margin-top: 1.5em; margin-bottom: 0.5em; color: rgb(3, 3, 3); }
.markdown-preview:not([data-use-github-style]) h1 { font-size: 2.4em; font-weight: 300; }
.markdown-preview:not([data-use-github-style]) h2 { font-size: 1.8em; font-weight: 400; }
.markdown-preview:not([data-use-github-style]) h3 { font-size: 1.5em; font-weight: 500; }
.markdown-preview:not([data-use-github-style]) h4 { font-size: 1.2em; font-weight: 600; }
.markdown-preview:not([data-use-github-style]) h5 { font-size: 1.1em; font-weight: 600; }
.markdown-preview:not([data-use-github-style]) h6 { font-size: 1em; font-weight: 600; }
.markdown-preview:not([data-use-github-style]) strong { color: rgb(3, 3, 3); }
.markdown-preview:not([data-use-github-style]) del { color: rgb(126, 126, 126); }
.markdown-preview:not([data-use-github-style]) a, .markdown-preview:not([data-use-github-style]) a code { color: black; }
.markdown-preview:not([data-use-github-style]) img { max-width: 100%; }
.markdown-preview:not([data-use-github-style]) > p { margin-top: 0px; margin-bottom: 1.5em; }
.markdown-preview:not([data-use-github-style]) > ul, .markdown-preview:not([data-use-github-style]) > ol { margin-bottom: 1.5em; }
.markdown-preview:not([data-use-github-style]) blockquote { margin: 1.5em 0px; font-size: inherit; color: rgb(126, 126, 126); border-color: rgb(214, 214, 214); border-width: 4px; }
.markdown-preview:not([data-use-github-style]) hr { margin: 3em 0px; border-top: 2px dashed rgb(214, 214, 214); background: none; }
.markdown-preview:not([data-use-github-style]) table { margin: 1.5em 0px; }
.markdown-preview:not([data-use-github-style]) th { color: rgb(3, 3, 3); }
.markdown-preview:not([data-use-github-style]) th, .markdown-preview:not([data-use-github-style]) td { padding: 0.66em 1em; border: 1px solid rgb(214, 214, 214); }
.markdown-preview:not([data-use-github-style]) code { color: rgb(3, 3, 3); background-color: rgb(240, 240, 240); }
.markdown-preview:not([data-use-github-style]) pre.editor-colors { margin: 1.5em 0px; padding: 1em; font-size: 0.92em; border-radius: 3px; background-color: rgb(245, 245, 245); }
.markdown-preview:not([data-use-github-style]) kbd { color: rgb(3, 3, 3); border-width: 1px 1px 2px; border-style: solid; border-color: rgb(214, 214, 214) rgb(214, 214, 214) rgb(199, 199, 199); border-image: initial; background-color: rgb(240, 240, 240); }
.markdown-preview[data-use-github-style] { font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif; line-height: 1.6; word-wrap: break-word; padding: 30px; font-size: 16px; color: rgb(51, 51, 51); background-color: rgb(255, 255, 255); overflow: scroll; }
.markdown-preview[data-use-github-style] > :first-child { margin-top: 0px !important; }
.markdown-preview[data-use-github-style] > :last-child { margin-bottom: 0px !important; }
.markdown-preview[data-use-github-style] a:not([href]) { color: inherit; text-decoration: none; }
.markdown-preview[data-use-github-style] .absent { color: rgb(204, 0, 0); }
.markdown-preview[data-use-github-style] .anchor { position: absolute; top: 0px; left: 0px; display: block; padding-right: 6px; padding-left: 30px; margin-left: -30px; }
.markdown-preview[data-use-github-style] .anchor:focus { outline: none; }
.markdown-preview[data-use-github-style] h1, .markdown-preview[data-use-github-style] h2, .markdown-preview[data-use-github-style] h3, .markdown-preview[data-use-github-style] h4, .markdown-preview[data-use-github-style] h5, .markdown-preview[data-use-github-style] h6 { position: relative; margin-top: 1em; margin-bottom: 16px; font-weight: bold; line-height: 1.4; }
.markdown-preview[data-use-github-style] h1 .octicon-link, .markdown-preview[data-use-github-style] h2 .octicon-link, .markdown-preview[data-use-github-style] h3 .octicon-link, .markdown-preview[data-use-github-style] h4 .octicon-link, .markdown-preview[data-use-github-style] h5 .octicon-link, .markdown-preview[data-use-github-style] h6 .octicon-link { display: none; color: rgb(0, 0, 0); vertical-align: middle; }
.markdown-preview[data-use-github-style] h1:hover .anchor, .markdown-preview[data-use-github-style] h2:hover .anchor, .markdown-preview[data-use-github-style] h3:hover .anchor, .markdown-preview[data-use-github-style] h4:hover .anchor, .markdown-preview[data-use-github-style] h5:hover .anchor, .markdown-preview[data-use-github-style] h6:hover .anchor { padding-left: 8px; margin-left: -30px; text-decoration: none; }
.markdown-preview[data-use-github-style] h1:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h2:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h3:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h4:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h5:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h6:hover .anchor .octicon-link { display: inline-block; }
.markdown-preview[data-use-github-style] h1 tt, .markdown-preview[data-use-github-style] h2 tt, .markdown-preview[data-use-github-style] h3 tt, .markdown-preview[data-use-github-style] h4 tt, .markdown-preview[data-use-github-style] h5 tt, .markdown-preview[data-use-github-style] h6 tt, .markdown-preview[data-use-github-style] h1 code, .markdown-preview[data-use-github-style] h2 code, .markdown-preview[data-use-github-style] h3 code, .markdown-preview[data-use-github-style] h4 code, .markdown-preview[data-use-github-style] h5 code, .markdown-preview[data-use-github-style] h6 code { font-size: inherit; }
.markdown-preview[data-use-github-style] h1 { padding-bottom: 0.3em; font-size: 2.25em; line-height: 1.2; border-bottom: 1px solid rgb(238, 238, 238); }
.markdown-preview[data-use-github-style] h1 .anchor { line-height: 1; }
.markdown-preview[data-use-github-style] h2 { padding-bottom: 0.3em; font-size: 1.75em; line-height: 1.225; border-bottom: 1px solid rgb(238, 238, 238); }
.markdown-preview[data-use-github-style] h2 .anchor { line-height: 1; }
.markdown-preview[data-use-github-style] h3 { font-size: 1.5em; line-height: 1.43; }
.markdown-preview[data-use-github-style] h3 .anchor { line-height: 1.2; }
.markdown-preview[data-use-github-style] h4 { font-size: 1.25em; }
.markdown-preview[data-use-github-style] h4 .anchor { line-height: 1.2; }
.markdown-preview[data-use-github-style] h5 { font-size: 1em; }
.markdown-preview[data-use-github-style] h5 .anchor { line-height: 1.1; }
.markdown-preview[data-use-github-style] h6 { font-size: 1em; color: rgb(119, 119, 119); }
.markdown-preview[data-use-github-style] h6 .anchor { line-height: 1.1; }
.markdown-preview[data-use-github-style] p, .markdown-preview[data-use-github-style] blockquote, .markdown-preview[data-use-github-style] ul, .markdown-preview[data-use-github-style] ol, .markdown-preview[data-use-github-style] dl, .markdown-preview[data-use-github-style] table, .markdown-preview[data-use-github-style] pre { margin-top: 0px; margin-bottom: 16px; }
.markdown-preview[data-use-github-style] hr { height: 4px; padding: 0px; margin: 16px 0px; background-color: rgb(231, 231, 231); border: 0px none; }
.markdown-preview[data-use-github-style] ul, .markdown-preview[data-use-github-style] ol { padding-left: 2em; }
.markdown-preview[data-use-github-style] ul.no-list, .markdown-preview[data-use-github-style] ol.no-list { padding: 0px; list-style-type: none; }
.markdown-preview[data-use-github-style] ul ul, .markdown-preview[data-use-github-style] ul ol, .markdown-preview[data-use-github-style] ol ol, .markdown-preview[data-use-github-style] ol ul { margin-top: 0px; margin-bottom: 0px; }
.markdown-preview[data-use-github-style] li > p { margin-top: 16px; }
.markdown-preview[data-use-github-style] dl { padding: 0px; }
.markdown-preview[data-use-github-style] dl dt { padding: 0px; margin-top: 16px; font-size: 1em; font-style: italic; font-weight: bold; }
.markdown-preview[data-use-github-style] dl dd { padding: 0px 16px; margin-bottom: 16px; }
.markdown-preview[data-use-github-style] blockquote { padding: 0px 15px; color: rgb(119, 119, 119); border-left: 4px solid rgb(221, 221, 221); }
.markdown-preview[data-use-github-style] blockquote > :first-child { margin-top: 0px; }
.markdown-preview[data-use-github-style] blockquote > :last-child { margin-bottom: 0px; }
.markdown-preview[data-use-github-style] table { display: block; width: 100%; overflow: auto; word-break: keep-all; }
.markdown-preview[data-use-github-style] table th { font-weight: bold; }
.markdown-preview[data-use-github-style] table th, .markdown-preview[data-use-github-style] table td { padding: 6px 13px; border: 1px solid rgb(221, 221, 221); }
.markdown-preview[data-use-github-style] table tr { background-color: rgb(255, 255, 255); border-top: 1px solid rgb(204, 204, 204); }
.markdown-preview[data-use-github-style] table tr:nth-child(2n) { background-color: rgb(248, 248, 248); }
.markdown-preview[data-use-github-style] img { max-width: 100%; box-sizing: border-box; }
.markdown-preview[data-use-github-style] .emoji { max-width: none; }
.markdown-preview[data-use-github-style] span.frame { display: block; overflow: hidden; }
.markdown-preview[data-use-github-style] span.frame > span { display: block; float: left; width: auto; padding: 7px; margin: 13px 0px 0px; overflow: hidden; border: 1px solid rgb(221, 221, 221); }
.markdown-preview[data-use-github-style] span.frame span img { display: block; float: left; }
.markdown-preview[data-use-github-style] span.frame span span { display: block; padding: 5px 0px 0px; clear: both; color: rgb(51, 51, 51); }
.markdown-preview[data-use-github-style] span.align-center { display: block; overflow: hidden; clear: both; }
.markdown-preview[data-use-github-style] span.align-center > span { display: block; margin: 13px auto 0px; overflow: hidden; text-align: center; }
.markdown-preview[data-use-github-style] span.align-center span img { margin: 0px auto; text-align: center; }
.markdown-preview[data-use-github-style] span.align-right { display: block; overflow: hidden; clear: both; }
.markdown-preview[data-use-github-style] span.align-right > span { display: block; margin: 13px 0px 0px; overflow: hidden; text-align: right; }
.markdown-preview[data-use-github-style] span.align-right span img { margin: 0px; text-align: right; }
.markdown-preview[data-use-github-style] span.float-left { display: block; float: left; margin-right: 13px; overflow: hidden; }
.markdown-preview[data-use-github-style] span.float-left span { margin: 13px 0px 0px; }
.markdown-preview[data-use-github-style] span.float-right { display: block; float: right; margin-left: 13px; overflow: hidden; }
.markdown-preview[data-use-github-style] span.float-right > span { display: block; margin: 13px auto 0px; overflow: hidden; text-align: right; }
.markdown-preview[data-use-github-style] code, .markdown-preview[data-use-github-style] tt { padding: 0.2em 0px; margin: 0px; font-size: 85%; background-color: rgba(0, 0, 0, 0.04); border-radius: 3px; }
.markdown-preview[data-use-github-style] code::before, .markdown-preview[data-use-github-style] tt::before, .markdown-preview[data-use-github-style] code::after, .markdown-preview[data-use-github-style] tt::after { letter-spacing: -0.2em; content: " "; }
.markdown-preview[data-use-github-style] code br, .markdown-preview[data-use-github-style] tt br { display: none; }
.markdown-preview[data-use-github-style] del code { text-decoration: inherit; }
.markdown-preview[data-use-github-style] pre > code { padding: 0px; margin: 0px; font-size: 100%; word-break: normal; white-space: pre; background: transparent; border: 0px; }
.markdown-preview[data-use-github-style] .highlight { margin-bottom: 16px; }
.markdown-preview[data-use-github-style] .highlight pre, .markdown-preview[data-use-github-style] pre { padding: 16px; overflow: auto; font-size: 85%; line-height: 1.45; background-color: rgb(247, 247, 247); border-radius: 3px; }
.markdown-preview[data-use-github-style] .highlight pre { margin-bottom: 0px; word-break: normal; }
.markdown-preview[data-use-github-style] pre { word-wrap: normal; }
.markdown-preview[data-use-github-style] pre code, .markdown-preview[data-use-github-style] pre tt { display: inline; max-width: initial; padding: 0px; margin: 0px; overflow: initial; line-height: inherit; word-wrap: normal; background-color: transparent; border: 0px; }
.markdown-preview[data-use-github-style] pre code::before, .markdown-preview[data-use-github-style] pre tt::before, .markdown-preview[data-use-github-style] pre code::after, .markdown-preview[data-use-github-style] pre tt::after { content: normal; }
.markdown-preview[data-use-github-style] kbd { display: inline-block; padding: 3px 5px; font-size: 11px; line-height: 10px; color: rgb(85, 85, 85); vertical-align: middle; background-color: rgb(252, 252, 252); border-width: 1px; border-style: solid; border-color: rgb(204, 204, 204) rgb(204, 204, 204) rgb(187, 187, 187); border-image: initial; border-radius: 3px; box-shadow: rgb(187, 187, 187) 0px -1px 0px inset; }
.markdown-preview[data-use-github-style] a { color: rgb(51, 122, 183); }
.markdown-preview[data-use-github-style] code { color: inherit; }
.markdown-preview[data-use-github-style] pre.editor-colors { padding: 0.8em 1em; margin-bottom: 1em; font-size: 0.85em; border-radius: 4px; overflow: auto; }
.markdown-preview pre.editor-colors { user-select: auto; }
.scrollbars-visible-always .markdown-preview pre.editor-colors .vertical-scrollbar, .scrollbars-visible-always .markdown-preview pre.editor-colors .horizontal-scrollbar { visibility: hidden; }
.scrollbars-visible-always .markdown-preview pre.editor-colors:hover .vertical-scrollbar, .scrollbars-visible-always .markdown-preview pre.editor-colors:hover .horizontal-scrollbar { visibility: visible; }
.markdown-preview .task-list-item input[type="checkbox"] { position: absolute; margin: 0.25em 0px 0px -1.4em; }
.markdown-preview .task-list-item { list-style-type: none; }
.bracket-matcher .region {
  border-bottom: 1px dotted lime;
  position: absolute;
}
.line-number.bracket-matcher.bracket-matcher {
  color: #555;
  background-color: rgba(255, 255, 134, 0.34);
}

.spell-check-misspelling .region {
  border-bottom: 2px dotted rgba(255, 51, 51, 0.75);
}
.spell-check-corrections {
  width: 25em !important;
}

pre.editor-colors.editor .gutter .line-number.line-number-green {
  color: #fff;
  background-color: #718C00;
}
pre.editor-colors.editor .gutter .line-number.line-number-blue {
  color: #fff;
  background-color: #96CBFE;
}
pre.editor-colors.editor .gutter .line-number.line-number-red {
  color: #fff;
  background-color: #D14;
}
.line.line-green {
  background-color: #718C00;
}
.line.line-blue {
  background-color: #96CBFE;
}
.line.line-red {
  background-color: #D14;
}

pre.editor-colors {
  background-color: white;
  color: #555;
}
pre.editor-colors .invisible-character {
  color: rgba(85, 85, 85, 0.2);
}
pre.editor-colors .indent-guide {
  color: rgba(85, 85, 85, 0.2);
}
pre.editor-colors .wrap-guide {
  background-color: rgba(85, 85, 85, 0.2);
}
pre.editor-colors .gutter {
  color: #555;
  background: white;
}
pre.editor-colors .gutter .line-number.folded,
pre.editor-colors .gutter .line-number:after,
pre.editor-colors .fold-marker:after {
  color: #e87b00;
}
pre.editor-colors .invisible {
  color: #555;
}
pre.editor-colors .selection .region {
  background-color: #e1e1e1;
}
pre.editor-colors .bracket-matcher .region {
  background-color: #C9C9C9;
  opacity: .7;
  border-bottom: 0 none;
}
pre.editor-colors.is-focused .cursor {
  border-color: black;
}
pre.editor-colors.is-focused .selection .region {
  background-color: #afc4da;
}
pre.editor-colors.is-focused .line-number.cursor-line-no-selection,
pre.editor-colors.is-focused .line.cursor-line {
  background-color: rgba(255, 255, 134, 0.34);
}
.syntax--comment {
  color: #999988;
  font-style: italic;
}
.syntax--string {
  color: #D14;
}
.syntax--string .syntax--source,
.syntax--string .syntax--meta.syntax--embedded.syntax--line {
  color: #5A5A5A;
}
.syntax--string .syntax--punctuation.syntax--section.syntax--embedded {
  color: #920B2D;
}
.syntax--string .syntax--punctuation.syntax--section.syntax--embedded .syntax--source {
  color: #920B2D;
}
.syntax--constant.syntax--numeric {
  color: #D14;
}
.syntax--constant.syntax--language {
  color: #606aa1;
}
.syntax--constant.syntax--character,
.syntax--constant.syntax--other {
  color: #606aa1;
}
.syntax--constant.syntax--symbol {
  color: #990073;
}
.syntax--constant.syntax--numeric.syntax--line-number.syntax--find-in-files .syntax--match {
  color: rgba(143, 190, 0, 0.63);
}
.syntax--variable {
  color: #008080;
}
.syntax--variable.syntax--parameter {
  color: #606aa1;
}
.syntax--keyword {
  color: #222;
  font-weight: bold;
}
.syntax--keyword.syntax--unit {
  color: #445588;
}
.syntax--keyword.syntax--special-method {
  color: #0086B3;
}
.syntax--storage {
  color: #222;
}
.syntax--storage.syntax--type {
  color: #222;
}
.syntax--entity.syntax--name.syntax--class {
  text-decoration: underline;
  color: #606aa1;
}
.syntax--entity.syntax--other.syntax--inherited-class {
  text-decoration: underline;
  color: #606aa1;
}
.syntax--entity.syntax--name.syntax--function {
  color: #900;
}
.syntax--entity.syntax--name.syntax--tag {
  color: #008080;
}
.syntax--entity.syntax--other.syntax--attribute-name {
  color: #458;
  font-weight: bold;
}
.syntax--entity.syntax--name.syntax--filename.syntax--find-in-files {
  color: #E6DB74;
}
.syntax--support.syntax--constant,
.syntax--support.syntax--function,
.syntax--support.syntax--type {
  color: #458;
}
.syntax--support.syntax--class {
  color: #008080;
}
.syntax--invalid {
  color: #F8F8F0;
  background-color: #00A8C6;
}
.syntax--invalid.syntax--deprecated {
  color: #F8F8F0;
  background-color: #8FBE00;
}
.syntax--meta.syntax--structure.syntax--dictionary.syntax--json > .syntax--string.syntax--quoted.syntax--double.syntax--json,
.syntax--meta.syntax--structure.syntax--dictionary.syntax--json > .syntax--string.syntax--quoted.syntax--double.syntax--json .syntax--punctuation.syntax--string {
  color: #000080;
}
.syntax--meta.syntax--structure.syntax--dictionary.syntax--value.syntax--json > .syntax--string.syntax--quoted.syntax--double.syntax--json {
  color: #d14;
}
.syntax--meta.syntax--diff,
.syntax--meta.syntax--diff.syntax--header {
  color: #75715E;
}
.syntax--css.syntax--support.syntax--property-name {
  font-weight: bold;
  color: #333;
}
.syntax--css.syntax--constant {
  color: #099;
}
.syntax--source.syntax--gfm {
  color: #444;
}
.syntax--gfm .syntax--markup.syntax--heading {
  color: #111;
}
.syntax--gfm .syntax--link {
  color: #888;
}
.syntax--gfm .syntax--variable.syntax--list {
  color: #888;
}
.syntax--markdown .syntax--paragraph {
  color: #444;
}
.syntax--markdown .syntax--heading {
  color: #111;
}
.syntax--markdown .syntax--link {
  color: #888;
}
.syntax--markdown .syntax--link .syntax--string {
  color: #888;
}
</style>
        </head>
        <body class='markdown-preview' data-use-github-style><h1 id="demultiplexing-with-alademux">Demultiplexing with alademux</h1>
<p>The tool <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">alademux</code> is meant to allow you to customize your demultiplexing
needs in a la carte manner, hence the name. What follows are examples of how to
demultiplex the 5 types of commonly encountered scenarios in the bioinformatics
core.</p>
<p>Log in to hci-zion. The software <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">alademux</code> software was built and tested
 on python3.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load python3</span></div></pre>
<h2 id="overview">Overview</h2>
<p>In this tutorial assume the git repository has been cloned to your home directory.
Review the usage of the main <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">alademux</code> command:</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/alademux.py <span class="syntax--entity syntax--other syntax--attribute-name">-h</span></span></div></pre>
<p>You will see the basis of demultiplexing starts with a required argument of the
Illumina instrument run id.  Every other argument is optional; for example,
you may specify a subset of lanes to demultiple. By default the type of
demultiplexing defaults to the standard scenario.  One can override the
input and output file paths. Importantly, the last argument allows you to
pass any valid <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">bcl2fastq</code> argument to the demultiplexing run.</p>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">usage: alademux.py [-h] -r run [-l [lane [lane ...]]]</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                   </span><span class="syntax--meta syntax--paragraph syntax--text">[-t [{standard,10x,10x-atac,patchpcr}]] [-i [RUN_PATH]]</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text"><span class="leading-whitespace">                   </span>[-o [OUT_PATH]] [-b ...]</span></span></div><div class="line"></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">optional arguments:</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">  </span><span class="syntax--meta syntax--paragraph syntax--text">-h, --help            show this help message and exit</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text"><span class="leading-whitespace">  </span>-r run, --run_id run  Name of Illumina Instrument Run (e.g.</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                        </span><span class="syntax--meta syntax--paragraph syntax--text">190624_A00421_0081_AHC7G3DRXX)</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">  </span><span class="syntax--meta syntax--paragraph syntax--text">-l [lane [lane ...]], --lanes [lane [lane ...]]</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                        </span><span class="syntax--meta syntax--paragraph syntax--text">List of lanes to demultiplex. e.g. -l 1 4 8 or -l 2</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">  </span><span class="syntax--meta syntax--paragraph syntax--text">-t [{standard,10x,10x-atac,patchpcr}], --type [{standard,10x,10x-atac,patchpcr}]</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                        </span><span class="syntax--meta syntax--paragraph syntax--text">Library type, valid options include: standard, 10x,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text"><span class="leading-whitespace">                        </span>10x-atac, patchpcr</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">  </span><span class="syntax--meta syntax--paragraph syntax--text">-i [RUN_PATH], --run_path [RUN_PATH]</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                        </span><span class="syntax--meta syntax--paragraph syntax--text">Path to directory with Illumina Instrument Run</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">  </span><span class="syntax--meta syntax--paragraph syntax--text">-o [OUT_PATH], --out_path [OUT_PATH]</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                        </span><span class="syntax--meta syntax--paragraph syntax--text">Path to demultiplexing results</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">  </span><span class="syntax--meta syntax--paragraph syntax--text">-b ..., --bcl2fastq ...</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="leading-whitespace">                        </span><span class="syntax--meta syntax--paragraph syntax--text">Any argument to pass to bcl2fastq. Note, these args</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text"><span class="leading-whitespace">                        </span>are not sanitized.</span></span></div></pre><h2 id="preview-run-contents">Preview run contents</h2>
<p>Suppose we want to demultiplex a recent NovaSeq run. Preview the library types
on a (library,project) basis.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/preview_demux.py 190920_A00421_0112_BHKTM3DMXX</span></div></pre>
<p>The preview shows</p>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:12 Flow cell configuration:</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:12 Number | NumCycles | IndexRead</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:12 1      | 151       | N</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:12 2      | 10       | Y</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:12 3      | 10       | Y</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:12 4      | 151       | N</span></span></div><div class="line"></div><div class="line"></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">&gt;Lane,Application,Sample_Project</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">1,10X Genomics Single Cell 3' Gene Expression Library Prep v2,17385R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">1,10X Genomics Single Cell 3' Gene Expression Library Prep v3,17334R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">1,10X Genomics Single Cell Mouse T Cell V(D)J Enrichment,17350R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,Illumina TruSeq DNA Nano LIbrary Prep (450 bp mean insert size) with UDI,16274R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,Illumina TruSeq DNA Nano LIbrary Prep (450 bp mean insert size) with UDI,17353R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,Illumina TruSeq Stranded mRNA Library Prep (RIN 8-10) with UDI,17354R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,Illumina TruSeq Stranded Total RNA Library Prep Ribo-Zero Gold (human, mouse, rat etc) ,16180R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,NEBNext Ultra II Directional RNA Library Prep with rRNA Depletion Kit (human,mouse,rat),16093R,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,NEBNext Ultra II Directional RNA Library Prep with rRNA Depletion Kit (human,mouse,rat),16298R,</span></span></div></pre><p>In this first example, we encounter a run with 10x Genomics libraries on lane 1
as well as other Illumina/NEB libraries fitting the standard scenario on lane 2. Let's explore how to demultiplex lane 2 first.</p>
<h2 id="standard-demultiplex">Standard demultiplex</h2>
<p>Set up the standard demultiplexing scripts for lane 2 with the following command.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/alademux.py <span class="syntax--entity syntax--other syntax--attribute-name">-r</span> 190920_A00421_0112_BHKTM3DMXX <span class="syntax--entity syntax--other syntax--attribute-name">-l</span> 2</span></div></pre>
<p>The output will give you the commands to start demultiplexing from the
target output directory.</p>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Start demultiplexing :</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">cd /path/to/demux/190920_A00421_0112_BHKTM3DMXX/20190925-173229</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">nohup ./demux.sh &amp;</span></span></div></pre><p>The reason for not starting the demultiplexing in a single command is if
the user wants to inspection the script or sample sheet before launching the demultiplexing job.</p>
<h2 id="10x-genomics-libraries">10x Genomics Libraries</h2>
<h3 id="53-expr-and-vdj">[5'/3' expr. and V(D)J]</h3>
<p>The 10x 5',3' and V(D)J libraries are handled a little differently than the
standard scenario. Trim the reads based on the 10x Genomics docs <a href="(https://support.10xgenomics.com/permalink/3IQFKIvEuskMoEWkWUis2s)">here</a> and <a href="https://support.10xgenomics.com/single-cell-vdj/sequencing/doc/specifications-sequencing-requirements-for-single-cell-vdj">here</a>:</p>
<ul>
<li><strong>R1</strong> Keep the first 28 bases. The 3' gene expression libraries (v3 chemistry) uses 28 bases for R1, 5' gene expression and V(D)J uses  26 bases.</li>
<li><strong>R2</strong> Keep 8 bp for the sample index.</li>
<li><strong>R3</strong> Do not use read; the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">alademux</code> software will add the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">--ignore-dual-index</code>
option automatically to the script.</li>
<li><strong>R4</strong> Use the full length except for the last cycle which is generally lower
quality.</li>
</ul>
<p>The <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">--use-bases-mask</code> argument is how the trimming instructions get passed
along. Recall these instructions are specific to the flow cell, which is why
we first preview the flow cell's contents to get the number of cycles correct.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/alademux.py <span class="syntax--entity syntax--other syntax--attribute-name">-r</span> 190920_A00421_0112_BHKTM3DMXX <span class="syntax--entity syntax--other syntax--attribute-name">-l</span> 1 \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-t</span> 10x \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-b</span> <span class="syntax--string">"--use-bases-mask='Y28n*,i8nn,n*,Y150n'"</span></span></div></pre>
<p>Behind the scenes, this command generates the sample sheet from GNomEx (see file GNomEx_SampleSheet.csv) and swaps the nucleotide based index with the 10x Genomics tag (see file SampleSheet.csv).</p>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Start demultiplexing :</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">cd /path/to/demux/190920_A00421_0112_BHKTM3DMXX/20190925-174609</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">nohup ./demux.sh &amp;</span></span></div></pre><p>Listing the files in the demultiplexing path you can see the demultiplexing
script as well as the 2 sample sheets.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">ls</span> /path/to/demux/190920_A00421_0112_BHKTM3DMXX/20190925-174609</span></div></pre>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">demuxer.sh  GNomEx_SampleSheet.csv  SampleSheet.csv</span></span></div></pre><h3 id="atac">ATAC</h3>
<p>While there are no recent 10x ATAC libraries to demonstrate the software,
we can still show what commands to run in this demultiplexing scenario.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/preview_demux.py 190710_A00421_0084_AHC7K3DRXX</span></div></pre>
<p>Because the run has been archived to tape, the flow cell configuration is
hidden from view.
å</p>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 18:05 Directory /path/to/illumina_runs/190710_A00421_0084_AHC7K3DRXX does not exist.</span></span></div><div class="line"></div><div class="line"></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Lane,Application,Sample_Project</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">1,10X Genomics Chromium Single Cell ATAC Library Prep,16164R1,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">1,10X Genomics Chromium Single Cell ATAC Library Prep,16237R1,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,10X Genomics Chromium Single Cell ATAC Library Prep,16164R1,</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2,10X Genomics Chromium Single Cell ATAC Library Prep,16237R1,</span></span></div></pre><p>Demultiplex by changing a few subtle options in the type argument and the trimming arguments as instructed by <a href="https://support.10xgenomics.com/single-cell-atac/sequencing/doc/specifications-sequencing-requirements-for-single-cell-atac">the 10x Genomics online docs for
ATAC libraries</a>.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/alademux.py <span class="syntax--entity syntax--other syntax--attribute-name">-r</span> 190710_A00421_0084_AHC7K3DRXX  \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-t</span> 10x-atac \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-b</span> <span class="syntax--string">"--use-bases-mask='Y50,I8,Y16,Y49'"</span></span></div></pre>
<h3 id="long-ranger">Long ranger</h3>
<p>If you have long ranger libraries,</p>
<ol>
<li><p>Change the demultiplexing type:</p>
<p><code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-t='10x-long'</code></p>
</li>
<li><p>Per the <a href="https://support.10xgenomics.com/genome-exome/sequencing/doc/specifications-sequencing-requirements-for-genome-and-exome">10x Genomics website</a> trim the reads:</p>
<p><code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-b "--use-bases-mask='Y150,I8,n*,Y150'"</code></p>
</li>
</ol>
<p>As of this writing, QC metrics are not generated for the long ranger results
and will not be reported to the lab.</p>
<h2 id="patch-pcr">Patch PCR</h2>
<p>A MiSeq run contains KT-Varley's Patch PCR libraries. Demultiplexing follows similar steps as before. First preview the run so we can trim  reads appropriately.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/preview_demux.py 190917_M00736_0336_MS8428226-300V2</span></div></pre>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:52 Flow cell configuration:</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:52 Number | NumCycles | IndexRead</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:52 1      | 151       | N</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:52 2      | 9       | Y</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:52 3      | 9       | Y</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">2019-09-25 17:52 4      | 151       | N</span></span></div><div class="line"></div><div class="line"></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Lane,Application,Sample_Project</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">1,Varley Patch PCR Target Enrichment,17400R,</span></span></div></pre><p>In this case, the script must have a 'y' for the <strong>R3</strong> read, to capture the UMI
sequence. This run is from a paired-end configuration, but sometimes Patch PCR
runs have a single end configuration.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/alademux.py <span class="syntax--entity syntax--other syntax--attribute-name">-r</span> 190917_M00736_0336_MS8428226-300V2 \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-t</span> patchpcr \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-b</span> <span class="syntax--entity syntax--other syntax--attribute-name">--use-bases-mask=</span><span class="syntax--string">'y150n,i8n,y9,y150n'</span></span></div></pre>
<h2 id="miseq-nano-run">MiSeq Nano Run</h2>
<p>These runs require you to specify a manual SampleSheet.csv provided by the HT-Genomics Sequencing group. Often, the core puts these runs in a different path than the standard Illumina runs, which requires us to specify the path.</p>
<p>If the run contains Nextera adapters (10 bp ), then you must specify the
<code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-n</code> flag.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">python3</span> alademux/alademux.py <span class="syntax--entity syntax--other syntax--attribute-name">-r</span> 190927_M05774_0084_MS8566788-050V2 \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-i</span> /path/to/nano \</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="leading-whitespace">  </span><span class="syntax--entity syntax--other syntax--attribute-name">-t</span> nano <span class="syntax--entity syntax--other syntax--attribute-name">-n</span></span></div></pre>
<p>This command will create the output directory, specify the correct adapter,
and create the correct script. You will see output like this:</p>
<pre class="editor-colors lang-text"><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Nano: No SampleSheet.csv being written. User must specify SampleSheet.csv</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Trimming Nextera adapters...</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">Start demultiplexing with the following commands:</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">cd /path/to/demux/190927_M05774_0084_MS8566788-050V2/20190930-220535</span></span></div><div class="line"><span class="syntax--text syntax--plain"><span class="syntax--meta syntax--paragraph syntax--text">nohup ./demuxer.sh &amp;</span></span></div></pre><p>You will still need to copy the SampleSheet.csv to the output directory.</p></body>
      </html>
